### PR TITLE
Change custom class & add type stubs

### PR DESCRIPTION
**Pull request by realerikrani**
Classmethods were not actually returning Keystore but the custom class itself.
The custom class's signature of "encrypt" was incompatible with supertype "Keystore".

Deleted one redundant method by using a modified class method.
Added missing "assert" keywords to default-values test and added a new test for creating an instance from private key.

### CODE REVIEW

File: test/cli/test_crypto_pbkdf2_keystore.py [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
 
 def test_it_sets_default_values():
     # when
     result = HorcruxPbkdf2Keystore()
 
     # then
-    result.index == 0
-    result.threshold == 0
-    result.shared_public_key == ""
-    result.shared_withdrawal_credentials == ""
```
Comment (realerikrani) [inline test/cli/test_crypto_pbkdf2_keystore.py:12]: Added missing asserts

File: cli/create.py [hunk 2]
```diff
@@ -295,7 +295,7 @@ def create(
     )
 
     # save horcrux private key to the keystore
-    keystore = create_keystore(
```
Comment (realerikrani) [inline cli/create.py:298]: Replaced `create_keystore` by `HorcruxPbkdf2Keystore.from_private_key`.

File: cli/create.py [hunk 3]
```diff
@@ -295,7 +295,7 @@ def create(
...
+    keystore = HorcruxPbkdf2Keystore.from_private_key(
```
Comment (tsudmi) [inline cli/create.py]: The `HorcruxPbkdf2Keystore` is used for encrypting the private key and storing it in the keystore file. I suggest renaming `from_private_key` to something else like `create`?

**Review by tsudmi — COMMENTED** [review]
Can we skip creating `eth2deposit` stubs for now?
Most of the functions in stubs we don't use as of today.

File: cli/create.py (hunk 3 above)
Comment (realerikrani) [reply cli/create.py]: Renamed in 06ac5c4.

**Comment by realerikrani** [comment]
> Can we skip creating eth2deposit stubs for now?

@tsudmi, I deleted the type stubs in 3cf3443,

**Review by tsudmi — APPROVED** [review]

**Pull request merged**